### PR TITLE
1.1.3

### LINKED ISSUE #1053: Vessels take a hard turn on out-of-timewarp SoI changes
Reported by @aexoden:
>
\<Aexoden> Has anyone else seen their trajectory be wildly recalculated upon crossing an SOI boundary? It only happens while in physics mode (so not at 5x+ time warp)
\<Aexoden> To be more specific, I'll be on what looks like a good free return trajectory to the Mun with something like a 16km Mun periapsis, but once it crosses into the Mun SOI, suddenly it's instead a 400km periapsis on the other side of the Mun. The green line shows almost a right angle change in trajectory

### PR DESCRIPTION
**Pull request by eggrobin**
* Port to 1.1.3;
* Use the bug-aware API to handle the bugs (don't ask);
* Fix #1053;
* Don't set timewarp to a low value on SoI change (that was annoying when fooling around L<sub>1</sub>).

[The improvement to `Orbit` drift mentioned in the change log](http://forum.kerbalspaceprogram.com/index.php?/developerarticles.html/kerbal-space-program-patch-113-is-now-available-r195/&page=1) also helps us truly get rid of stock gravity.

### CODE REVIEW

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 1]
```diff
@@ -236,15 +238,19 @@ where is_in_physics_bubble(vessel)
...
         refBody : vessel.orbit.referenceBody,
         UT      : universal_time);
   }
 
   private void AddToPhysicsBubble(Vessel vessel) {
-    Vector3d gravity =
-        FlightGlobals.getGeeForceAtPosition(vessel.findWorldCenterOfMass());
+    if (FlightIntegrator.GraviticForceMultiplier != 0) {
+      Log.Info("Killing stock gravity");
+      FlightIntegrator.GraviticForceMultiplier = 0;  // sic.
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: The "sic" should be on the first occurrence (in the if condition).

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 2]
```diff
@@ -263,11 +269,13 @@ where is_in_physics_bubble(vessel)
...
-        // compensate for the pos/vel synchronization bug.
         plugin_.SetVesselStateOffset(
             vessel_guid : vessel.id.ToString(),
-            from_parent : new QP{q = (XYZ)vessel.orbit.pos,
+            from_parent : new QP{
+                q = vessel.orbitDriver.offsetPosByAFrame
+                      ? (XYZ)(vessel.orbit.pos -
+                              vessel.orbit.vel * TimeWarp.fixedDeltaTime)
+                      : (XYZ)vessel.orbit.pos,
                                  p = (XYZ)vessel.orbit.vel});
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Bad indent for `p`.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 3]
```diff
@@ -659,8 +671,17 @@ where is_in_physics_bubble(vessel)
         return;
       }
       time_is_advancing_ = true;
-      if (has_inertial_physics_bubble_in_space()) {
+      if (has_inertial_physics_bubble_in_space() &&
+          (FlightGlobals.currentMainBody == previous_bubble_reference_body_ ||
+           previous_bubble_reference_body_ == null)) {
         ApplyToVesselsInPhysicsBubble(AddToPhysicsBubble);
+        previous_bubble_reference_body_ = FlightGlobals.currentMainBody;
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: This is a no-op if the first part of the condition is true.  Is this intended?

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 1 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 2 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 3 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Yes.

**Pull request merged**